### DISCUSSION TITLE
TypeScript detection not working properly in node.js scripts

### DISCUSSION THREAD

**Discussion by eugene-matvejev** [github-linguist/linguist#6692] [Classification]
## Describe the bug
Hello,

I mentor a person who work on side project
https://github.com/search?q=repo%3Aanna-liepina%2Fexplore-sa-node++language%3AJavaScript&type=code
but for some reason a lot of typescript files are not recognized as typescript

### Expected behaviour
files should be recognized because files are 100% typescript

### Related discussion
n/a

### Additional notes
demo link:
https://github.com/search?q=repo%3Aanna-liepina%2Fexplore-sa-node++language%3AJavaScript&type=code

**Comment by Alhadis** [comment]
The files are getting classified as Node.js programs because of the `#!/usr/bin/env node` hashbang. Linguist doesn't consider Node to be a TypeScript interpreter (nor should it), so I recommend using [an override][1] or a [Vim/Emacs-style modeline][2]:

[1]: https://github.com/github-linguist/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/docs/overrides.md#using-gitattributes
[2]: https://github.com/github-linguist/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/docs/overrides.md#using-emacs-or-vim-modelines

**Using an override:**
```gitattributes
# Add the following to your project's `.gitattributes` file
*.ts linguist-language=TypeScript
```

**Using modelines:**
```ts
#!/usr/bin/env node
// -*- mode: typescript; -*-

declare type Script = "to be this file's language";
```
